plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways-react.php

```diff
@@ -0,0 +1,81 @@
+<?php // @codingStandardsIgnoreLine.
+/**
+ * WooCommerce Checkout Settings
+ *
+ * @package WooCommerce\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+if ( class_exists( 'WC_Settings_Payment_Gateways', false ) ) {
```

Comment:
I maintained the old class name for potential compat issues (such as checking for existance of `WC_Settings_Payment_Gateways` class)

Reply:
Actually, this is erroring in lint. Perhaps not a good solution 🤔 

```
FILE: class-wc-settings-payment-gateways-react.php
-----------------------------------------------------------------------------------------------
FOUND 1 ERROR AND 0 WARNINGS AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------
 1 | ERROR | Class file names should be based on the class name with "class-" prepended. Expected class-wc-settings-payment-gateways.php, but found class-wc-settings-payment-gateways-react.php. (WordPress.Files.FileName.InvalidClassFileName)
-----------------------------------------------------------------------------------------------
```

Reply:
I think it will be better practice to give the class a unique name. However do you think 3PD might be checking for the existence of a `WC_Settings_Payment_Gateways` class?

Reply:
I've updated the name to `WC_Settings_Payment_Gateways_React` in 3d459a7  😅  

> However do you think 3PD might be checking for the existence of a `WC_Settings_Payment_Gateways` class?

I don't have any evidence for it, but I imagine they could. Though the extension points are filters and actions, so I think it may be unlikely. Also, 3PD is more likely to check for `WC_Settings_Page` class when extending the settings page, so it should be safe 🤞 

I'm open for a better alternative though!

Reply:
> it should be safe 

I think so too. I've also just now made a quick search for usage in https://github.com/woocommerce/all-plugins and there were no results 👍